xarray/core/duck_array_ops.py

```diff
@@ -53,6 +65,89 @@
 dask_available = module_available("dask")
 
 
+HANDLED_EXTENSION_ARRAY_FUNCTIONS: dict[Callable, Callable] = {}
+
+
+def implements(numpy_function):
+    """Register an __array_function__ implementation for MyArray objects."""
+
+    def decorator(func):
+        HANDLED_EXTENSION_ARRAY_FUNCTIONS[numpy_function] = func
+        return func
+
+    return decorator
+
+
+@implements(np.issubdtype)
+@dispatch
+def __extension_duck_array__issubdtype(
+    extension_array_dtype: T_ExtensionArray, other_dtype: DTypeLikeSave
+):
+    return False  # never want a function to think a categorical is a subtype of numpy
+
+
+@implements(np.broadcast_to)
+@dispatch
+def __extension_duck_array__broadcast(arr: T_ExtensionArray, shape: tuple):
```

Comment:
@benbovy speaking of the `indexing.py` file, I noticed the `PandasIndexingAdapter` returns a `np.array` version of the index in certain cases so that multi-dim opertaions (such as broadcast?) can work.  Should I do the same here?   I explicitly don't support this functionality here (see [point 1. on this comment](https://github.com/pydata/xarray/issues/5287#issuecomment-839079791) for why I made this decision).  I don't have strong feelings, but I am inclined to not support it by default, especially since with `plum.dispatch`, the behavior can be overridden.

Reply:
I don't have strong feelings either on this.